projectroles/views.py

```diff
@@ -1757,8 +1760,18 @@ def delete_assignment(self, request, instance):
             )
         # Delete object itself
         instance.delete()
-        # Remove project star from user if it exists
-        app_settings.delete('projectroles', 'project_star', project, user)
+
+        # Delete corresponding PROJECT_USER settings
+        if (
+            not project.get_role(user, inherited_only=True)
+            and project.type == PROJECT_TYPE_CATEGORY
+            and not RoleAssignment.objects.filter(
```

Comment:
Here we also have an issue. This query looks for only the immediate child project, but the category structure can be deep and a category may have multiple levels of children under it. You should use `Project.get_children()` instead.